platform/editor.mimelookup/src/org/netbeans/modules/editor/mimelookup/CreateRegistrationProcessor.java

```diff
@@ -145,7 +147,17 @@ private void process(Element toRegister, AnnotationMirror mimeRegistration) thro
             }
 
             instantiableClassOrMethod(toRegister, apiTE);
-            layer(toRegister).instanceFile("Editors" + mimeType + folder, null, null).position(position).stringvalue("instanceOf", processingEnv.getElementUtils().getBinaryName(apiTE).toString()).write();    //NOI18N
+            File f = layer(toRegister).instanceFile("Editors" + mimeType + folder, null, null);
+            if (position == Integer.MAX_VALUE && !"".equals(folder)) {
```

Comment:
The problem is that with MIME being hierarchical, a single `position` defined in `text/x-java` means, that if anyone registers a general service (in `/`):
- if `position` is not specified, that service faisl commit-validation as the sequence text/x-java/* + /* contains both defined and undefined positions
- if general service **does** define position, then all MIMEs **other than java** must start to provide positions.

So as discussed with @JaroslavTulach, it's far more safer to generate some positions always. 

The goal is to:
- generate position
- prefer more specific (lower in the folder hierarchy) registrations
- define _some_ stable order
- generate different numbers :-)

Note the exclusion of the `""` (none) subfolder: various services are registered in `Editors/` (or relevant MIME) directly which do NOT have position at the moment, and are registerd manually (entries in layers.xml, ...). Tons of changes would happen if positions would be generated by the annotation processor there. So this change only affect "named" service subfolders.

Reply:
Right, generate some positions always.